src/undo_log/undo_log.c

```
@@ -0,0 +1,734 @@
+		UndoOp *op = UNDOLOG_GET_ITEM (ctx->undo_log, i) ;
+		UndoUpdateOp *update_op = &op->update_op ;
+
+		if (update_op->entity_type == GETYPE_NODE) {
+			// regain attribute ownership
+			AttributeSet_TransferOwnership (*update_op->n.attributes,
+					update_op->set) ;
+
+			// free current entity attribute-set
+			AttributeSet_Free (update_op->n.attributes) ;
+
+			// restore entity original attribute-set
```

Cppcheck (2.20.0)

[error] Common realloc mistake

(memleakOnRealloc)